l10n_it_account/models/account_tax.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Copyright (C) 2013 Associazione OpenERP Italia
+#    (<http://www.openerp-italia.org>).
+#    Copyright (C) 2014 Agile Business Group sagl
+#    (<http://www.agilebg.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp.osv import osv
+from openerp.tools.translate import _
+
+
+class account_tax(osv.osv):
+    _inherit = 'account.tax'
+
+    def copy_data(self, cr, uid, id, default=None, context=None):
+        if default is None:
+            default = {}
+        tmp_default = dict(default, base_code_id='', tax_code_id='', ref_base_code_id='', ref_tax_code_id='')
+        return super(account_tax, self).copy_data(cr, uid, id, default=tmp_default, context=context)
+
+    def create(self, cr, uid, vals, context=None):
+
+        if vals['type_tax_use'] == 'purchase':
+            if 'base_code_id' in vals:
+                if vals['base_code_id']:
+                    if self.exist(cr, uid, 'base_code_id', vals['base_code_id']):
+                        raise osv.except_osv(_('Error!'),
+                                             _('Invoices: Base code already used in other purchase tax.'))
```

Comment:
Suggerirei anche di implementare il check 'a posteriori': all'interno dei metodi create e write, chiamare subito la ```super``` e controllare il record risultante. Se il vincolo non è soddisfatto , fare il raise ( quindi il sistema farà il rollback senza salvare nulla nel DB).
In questo modo il metodo che controlla la consistenza dei dati può essere uno solo e risulterà più chiaro e semplice da scrivere